### DISCUSSION TITLE
Why is the type for the `params` field different between Notification and Request structs?

### DISCUSSION THREAD

**Discussion by AArnott** [paritytech/jsonrpsee#1457] [Q&A]
The request and notification structs are defined as:

```rs
#[derive(Serialize, Deserialize, Debug, Clone)]
pub struct Request<'a> {
	/// JSON-RPC version.
	pub jsonrpc: TwoPointZero,
	/// Request ID
	#[serde(borrow)]
	pub id: Id<'a>,
	/// Name of the method to be invoked.
	#[serde(borrow)]
	pub method: Cow<'a, str>,
	/// Parameter values of the request.
	#[serde(borrow)]
	pub params: Option<Cow<'a, RawValue>>,
	/// The request's extensions.
	#[serde(skip)]
	pub extensions: Extensions,
}

#[derive(Serialize, Deserialize, Debug, Clone)]
pub struct Notification<'a, T> {
	/// JSON-RPC version.
	pub jsonrpc: TwoPointZero,
	/// Name of the method to be invoked.
	#[serde(borrow)]
	pub method: Cow<'a, str>,
	/// Parameter values of the request.
	pub params: T,
}
```

Why are the `params` fields defined as different types between these structs? I expect the two structs to be identical except for the `id` field.

I also find it odd that most of these message structs define their fields as public, but ErrorObject's fields are private.

**Comment by niklasad1** [comment]
IIRC, it's because how jsonrpsee is using subscriptions where params is required and testing it but I see no problem to unify them at  and just use `T` for params for both the structs with some extra some type alias for params.
